pkg/joinserver/http_interop.go

```diff
@@ -105,16 +105,20 @@ func (srv interopServer) JoinRequest(ctx context.Context, in *interop.JoinReq) (
 		Result: interop.Result{
 			ResultCode: interop.ResultSuccess,
 		},
-		Lifetime:     uint32(res.Lifetime / time.Second),
-		AppSKey:      (*interop.KeyEnvelope)(res.AppSKey),
-		SessionKeyID: interop.Buffer(res.SessionKeyId),
+		AppSKey:      (*interop.KeyEnvelope)(res.SessionKeys.AppSKey),
+		SessionKeyID: interop.Buffer(res.SessionKeys.SessionKeyId),
 	}
+
+	if res.Lifetime != nil {
+		ans.Lifetime = uint32(ttnpb.StdDurationOrZero(res.Lifetime) / time.Second)
```

Comment:
Why the `nil` check ? 0 divided by something is still 0, right ?

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/blob/5b38d7e6a5cea6210c88afef918e74bfd11a2f90/pkg/joinserver/joinserver.go#L552-L555
The standard JS implementation does not set this field and it's optional.

Reply:
Ah yes right